src/utils/worker/createResponseListener.ts

```diff
@@ -25,7 +25,8 @@ export function createResponseListener(context: SetupWorkerInternalContext) {
       return
     }
 
-    const response = new Response(responseJson.body, responseJson)
+    const responseBody = responseJson.status === 204 ? null : responseJson.body
```

Comment:
I found https://fetch.spec.whatwg.org/#statuses => IMHO let's check all null body statuses here:

```suggestion
    const responseBody = [101, 204, 205, 304].includes(responseJson.status) ? null : responseJson.body
```

if this is hot-path for perf optimization, maybe a `new Set(...)` in some util file can be used 😂

Reply:
That's a great suggestion, but also a great example of why reflecting a spec this way is a huge redundancy. I wish to find a different way to retrieve a 204 response body as `null`.

Reply:
yeah, no ideas about the clean way yet :/

Reply:
well.. I do have an idea - `responseJson.body === '' ? null : responseJson.body` - but totally not sure how that would break all edge cases with empty body in `200` and other responses 😬

Reply:
You know why here https://github.com/mswjs/msw/commit/40920214c1c22a06d640cc3a1006528e2ba6e679  is working ?

Reply:
What about the responses that have an empty string body? There must be a differentiation between them as those responses that have no body (`null`). With the logic above, both cases will result in `res.body` being `null`.

Reply:
yeah, this distinction between null and empty string is really strange, no idea what was the reason to make it different in the spec :(

Reply:
Can I  use this suggested change as a stop-gap for now? I'm using only `msw` for local development and it'd be nice to not have that warning in console (I'm using `patch-package` for these purposes).

Reply:
@comatory, I'd rather recommend you edit the `mockServiceWorker.js` with a similar change instead. Changing what lies in `node_modules` isn't a reliable way to fix anything, as the modules will be pruned once you re-install the dependencies.

Reply:
@kettanaito I see. However with [patch-package](https://www.npmjs.com/package/patch-package) the changes are re-applied after installation so it's safe (and actually recommended way) to do it. I'm fine with this solution, once this problem is fixed, I'll just remove the patch from my codebase.